contrib/src/databases.rs

```diff
@@ -43,23 +123,27 @@ impl Fairing for Databases {
     }
 
     fn on_attach(&self, mut rocket: Rocket) -> Result<Rocket, Rocket> {
+        let database_initializers = DatabaseInitializers::new();
+
         for database_config in rocket.config().clone().get_slice("databases").unwrap() {
+            let adapter_name = database_config["name"].as_str().unwrap();
+            let credentials = database_config["url"].as_str().unwrap();
+            let pool_size = database_config["pool_size"].as_integer().unwrap_or(10) as u32;
             match database_config["adapter"] {
-                String(ref adapter) => {
-                    if adapter == "diesel_sqlite" {
-                        rocket = rocket.manage(self.init_sqlite_pool(database_config["url"].as_str().unwrap()));
-                    } else if adapter == "rust-postgres" {
-                        rocket = rocket.manage(self.init_rust_pg_pool(database_config["url"].as_str().unwrap()));
-                    }
-                },
-                _ => return Err(rocket),
+                Value::String(ref adapter_type) => {
+                    rocket = database_initializers.pool_mappers[adapter_type](
+                        rocket,
+                        database_initializers.initializers[adapter_type].init_pool(
+                            credentials,
+                            pool_size
+                        ),
+                        adapter_name.to_string()
+                    );
```

Comment:
The formatting is a little weird here, but basically, there's a collection of closures based on adapter type that are run to reify the connection pool type in the collection of generic pools (casting `std::any::Any` to a concrete `Pool<T>`). This is intended to result in user extensibility.